app/api/security.py

```diff
@@ -0,0 +1,43 @@
+import os
+
+from fastapi import HTTPException, status
+from fastapi.security.utils import get_authorization_scheme_param
+from google.auth.exceptions import GoogleAuthError
+from google.auth.transport import requests
+from google.oauth2 import id_token
+
+AUTH_ENABLED = os.environ.get("NB_ENABLE_AUTH", "True").lower() == "true"
+CLIENT_ID = os.environ.get("NB_QUERY_CLIENT_ID", None)
+
+
+def check_client_id():
+    """Check if the CLIENT_ID environment variable is set."""
+    # By default, if CLIENT_ID is not provided to verify_oauth2_token,
+    # Google will simply skip verifying the audience claim of ID tokens.
+    # This however can be a security risk, so we mandate that CLIENT_ID is set.
+    if AUTH_ENABLED and CLIENT_ID is None:
+        raise ValueError(
+            "Authentication has been enabled (NB_ENABLE_AUTH) but the environment variable NB_QUERY_CLIENT_ID is not set. "
+            "Please set NB_QUERY_CLIENT_ID to the Neurobagel query tool client ID, to verify the audience claim of ID tokens."
```

Comment:
```suggestion
            "Please set NB_QUERY_CLIENT_ID to the Google client ID for your Neurobagel query tool deployment, to verify the audience claim of ID tokens."
```
As you said, we want folks to get their own app registered.